packages/shorebird_cli/test/src/shorebird_android_artifacts_test.dart

```diff
@@ -0,0 +1,347 @@
+import 'package:path/path.dart' as path;
+import 'package:shorebird_cli/src/shorebird_android_artifacts.dart';
+import 'package:shorebird_cli/src/third_party/flutter_tools/lib/flutter_tools.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group(ShorebirdAndroidArtifacts, () {
+    late Directory project;
+    late ShorebirdAndroidArtifacts shorebirdAndroidArtifacts;
+
+    setUp(() {
+      project = Directory.systemTemp.createTempSync();
+      shorebirdAndroidArtifacts = ShorebirdAndroidArtifacts();
+    });
+
+    group('when cannot find artifacts', () {
+      group('when no build folder exists', () {
+        test('throws ArtifactNotFoundException for aabs', () {
+          expect(
+            () => shorebirdAndroidArtifacts.findAppBundle(
+              project: project,
+              flavor: null,
+            ),
+            throwsA(isA<ArtifactNotFoundException>()),
+          );
+        });
+        test('throws ArtifactNotFoundException for apks', () {
+          expect(
+            () => shorebirdAndroidArtifacts.findApk(
+              project: project,
+              flavor: null,
+            ),
+            throwsA(isA<ArtifactNotFoundException>()),
+          );
+        });
+      });
+      group('when build folder exists but not the file', () {
+        test('throws ArtifactNotFoundException for aabs', () {
+          Directory(
+            path.join(
+              project.path,
+              'build',
+              'app',
+              'outputs',
+              'bundle',
+              'release',
+            ),
+          ).createSync(recursive: true);
```

Comment:
This could be in the group's setUp (instead of in the test itself)

Reply:
I am unsure, each of the tests uses a different path since one is about aabs and the other about apks, which lives in different places. So I fail to see the benefit.

Or maybe I am missing something?

Reply:
In this case, it's probably not necessary.